ci/unit_tests/test_pathology_tumor_detection.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import os
+import shutil
+import tempfile
+import unittest
+
+from monai.apps.utils import download_url
+from monai.bundle import ConfigWorkflow
+from parameterized import parameterized
+
+TEST_CASE_1 = [  # train
+    {
+        "bundle_root": "models/pathology_tumor_detection",
+        "region_size": [256, 256],
+        "num_epochs": 1,
+        "epochs": 1,
+        "train#dataloader#batch_size": 32,
+        "validate#dataloader#batch_size": 32,
+        "train#datalist#transform#func": "$lambda x: os.path.join(@dataset_dir, x + '.tiff')",
+        "validate#datalist#transform#func": "$lambda x: os.path.join(@dataset_dir, x + '.tiff')",
+    },
+    {
+        "bundle_root": "models/pathology_tumor_detection",
+        "dataloader#batch_size": 32,
+        "datalist#transform#func": "$lambda x: os.path.join(@dataset_dir, x + '.tiff')",
+    },
+]
+
+TIFF_INFO = {  # data to be downloaded for test
+    "url": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/CMU-1.tiff",
+    "filename": "CMU-1.tiff",
+    "hash_type": "sha256",
+    "hash_val": "73a7e89bc15576587c3d68e55d9bf92f09690280166240b48ff4b48230b13bcd",
+    "csv_info": ["CMU-1", 46000, 32914, 0, 0, 0, 0, 0, 0, 1, 1, 1],
+}
+
+
+class TestTumorDetection(unittest.TestCase):
+    def setUp(self):
+        self.dataset_dir = tempfile.mkdtemp()
+        # download sample tiff file
+        download_url(
+            url=TIFF_INFO["url"],
+            filepath=os.path.join(self.dataset_dir, TIFF_INFO["filename"]),
+            hash_type=TIFF_INFO["hash_type"],
+            hash_val=TIFF_INFO["hash_val"],
+        )
+        # prepare csv files
+        for csv_file in ["training.csv", "validation.csv"]:
+            with open(os.path.join("models/pathology_tumor_detection", csv_file), "w", newline="") as save_file:
+                wr = csv.writer(save_file)
+                wr.writerow(TIFF_INFO["csv_info"])
+
+        with open(os.path.join("models/pathology_tumor_detection", "testing.csv"), "w", newline="") as save_file:
+            wr = csv.writer(save_file)
+            wr.writerow(TIFF_INFO["csv_info"][:1])
+
+    def tearDown(self):
+        shutil.rmtree(self.dataset_dir)
+
+    @parameterized.expand([TEST_CASE_1])
+    def test_train_infer_config(self, override, override_infer):
```

Comment:
Looks like all these `test_train_infer_config` and `test_infer_config` are the same, do we need to extract these into functions in utils so that they can be called directly when other contributors need to write their own unit tests?

Reply:
Hi @KumoLiu , thanks for the suggestions. The functions are similar but may have some differences like:

1. Required files may be different, thus we still need to input many args into a utility function if defined.
2. Test ways may be different. Like in this bundle, we put train and infer together to avoid duplicating download, and in other bundles they may be in separate test functions.

Therefore, I think the code to define a config workflow should be written by contributors. However, I think we can define a function to simplify the initialize, check properties, run and finalize steps like:

```
def check_workflow(workflow: ConfigWorkflow, check_properties: bool = False):
    workflow.initialize()
    if check_properties is True:
        check_result = workflow.check_properties()
        if check_result is not None and len(check_result) > 0:
            raise ValueError(f"check properties for workflow failed: {check_result}")
    workflow.run()
    workflow.finalize()
```
Then contributor can call this function via:
```
check_workflow(trainer, False)
```

What do you think?

Reply:
Sounds good.

Reply:
Thanks, I updated in the latest commit